backend/src/chat.gateway.spec.ts

```diff
@@ -2,6 +2,9 @@ import { ConfigModule } from '@nestjs/config';
 import { Test, TestingModule } from '@nestjs/testing';
 import { getRepositoryToken } from '@nestjs/typeorm';
 import { Repository } from 'typeorm';
+import { AchievementsLogService } from './achievements-log/achievements-log.service';
+import { AchievementsLogRepository } from './achievements-log/repository/achievements-log.repository';
+import { AchievementRepository } from './achievements/repository/achievements.repository';
```

Comment:
I'm guessing the imports are due to some of the testing problems you mentioned, but it seems strange that the chat-gateway.spec would need them when the ChatGateway doesn't use them. I wonder if there is a problem with the way we're building our test modules

Reply:
I tend to add imports in a messy way to get rid of errors, so I'll test them separately tomorrow. I'm sure at least some of them are not necessary. At the same time i do feel like there are some strange behaviors, and I am not sure if the tests modules are to blame, or if it is just the fact that I often add lots of dependencies inside services.

Reply:
So here's the thing: AchivementsLog is a dependency of User, and Achievement is a dependency of Log, so not importing them here results in:
```
ChatGateway › should be defined

    Nest can't resolve dependencies of the UsersService (UserRepository, FriendshipRepository, BlockRepository, ?). Please make sure that the argument AchievementsLogRepository at index [3] is available in the RootTestModule context.

    Potential solutions:
    - If AchievementsLogRepository is a provider, is it part of the current RootTestModule?
    - If AchievementsLogRepository is exported from a separate @Module, is that module imported within RootTestModule?
      @Module({
        imports: [ /* the Module containing AchievementsLogRepository */ ]
      })
```
The other ones seem okay but I am not that confident about the E2E tests.
If you think of something to improve the situation let me know.

Reply:
Okay, I'll look at one thing tomorrow to see if I made a mistake in the creation of the ChatGateway, but if we can't find the solution then at least it works like this